### PR TITLE
Update docs to use Hint object

### PR DESCRIPTION
**Pull request by adinauer**
Update docs to reflect https://github.com/getsentry/sentry-java/pull/2045

### CODE REVIEW

File: src/includes/configuration/before-send-fingerprint/java.spring-boot.mdx [hunk 1]
```diff
@@ -1,12 +1,13 @@
 ```java
 import io.sentry.SentryEvent;
 import io.sentry.SentryOptions;
+import io.sentry.hints.Hint
```
Comment (adinauer) [inline src/includes/configuration/before-send-fingerprint/java.spring-boot.mdx:4]: Should we move `Hint` to `io.sentry` package so devs don't have to deal with different packages? Or is it OK like this?

**Comment by adinauer** [comment]
Do we also need docs for manipulating attachments via hints? If so where do we put them?

File: src/includes/configuration/before-send-fingerprint/java.spring-boot.mdx (hunk 1 above)
Comment (adinauer) [reply src/includes/configuration/before-send-fingerprint/java.spring-boot.mdx:4]: PR open https://github.com/getsentry/sentry-java/pull/2064

File: src/platforms/android/migration.mdx [hunk 2]
```diff
@@ -27,7 +27,7 @@ description: "Migrating between versions of Sentry's SDK for Android."
   * `IHandler` is now `MainLooperHandler` only.
 * `ISpan` now has higher precision using the `System#nanoTime` instead of milliseconds.
 
-* Hints changed its type from `Object` to `Map<String, Object>`
+* Hints changed its type from `Object` to `Hint`
```
[the remark this answers is not in the document]
Comment (adinauer) [reply src/platforms/android/migration.mdx]: Split out into https://github.com/getsentry/sentry-docs/pull/5068

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
@adinauer this can also be merged since its GA.

**Pull request merged**